### PR TITLE
Feature/1818/translation fuctionality

### PR DESCRIPTION
**Pull request by D1nviz**
Added translation functionality
image
Video how it works. (After deleting the language from localStorage, the language will be selected from the object navigator (preferences browser language) on the next launch of the site)

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/118522791/25c5ca6d-ca0a-45bd-bca6-e38dae65d749

- I added language additions to local storage.
- I found several bugs with layout and translations, which I will divide into other issues in the future. 
- Also fixed some tests that were failing
- Added an error message if the language cannot be changed

### CODE REVIEW

File: src/plugins/i18n.ts [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import i18n from 'i18next'
 import { initReactI18next } from 'react-i18next'
 import resources from '~/constants/translations'
+import { getFromLocalStorage } from '~/services/local-storage-service'
+
+const initialLanguage = getFromLocalStorage('language') || 'en'
```
[the remark this answers is not in the document]
Comment (D1nviz) [reply src/plugins/i18n.ts]: Yes, i did. But to be honest, I forgot about it. Thanks, that's a really cool idea, I've added that feature

File: src/containers/layout/language-menu/LanguageMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        })
+      })
+  }
+
+  const menuItems = languageMenuConstants.map(({ language, languageCode }) => {
+    return (
+      <MenuItem
+        key={languageCode}
+        onClick={() => handleLanguageChange(languageCode)}
+        sx={i18n.language === languageCode ? styles.active : {}}
```
[the remark this answers is not in the document]
Comment (D1nviz) [reply src/containers/layout/language-menu/LanguageMenu.tsx]: You're right, i fixed it

File: src/containers/navigation-icons/user-icons/UserIcons.tsx [hunk 3]
```diff
@@ -21,16 +23,25 @@ interface UserIconsProps {
 const UserIcons: FC<UserIconsProps> = ({ setSidebarOpen }) => {
   const [notificationItems, setNotificationItems] =
     useState<Notification[]>(mockNotifications)
-  const [menuAnchorEl, setMenuAnchorEl] = useState<HTMLElement | null>(null)
-  const [notificationsAnchor, setNotificationsAnchor] =
-    useState<HTMLElement | null>(null)
-  const anchorRef = useRef<HTMLDivElement | null>(null)
+  const {
+    anchorEl: accountenuAnchorEl,
```
Comment (asya982) [inline src/containers/navigation-icons/user-icons/UserIcons.tsx:27]:
```suggestion
    anchorEl: accountMenuAnchorEl,
```
I think it's a typo here

Comment (D1nviz) [reply src/containers/navigation-icons/user-icons/UserIcons.tsx]: Your right. Fixed

**Review by mhevyk — APPROVED** [review]

**Review by iva-stasia — APPROVED** [review]
I guess we should create an issue to check if everything looks coorect depending on the language and avoid such cases (My courses page):
image
Or you can check it in your PR.

File: src/containers/layout/language-menu/LanguageMenu.constants.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export const languageMenuConstants = [
+  {
+    languageCode: 'en',
+    language: 'English'
+  },
+  {
+    languageCode: 'ua',
+    language: 'Українська'
+  }
+]
```
Comment (byavdoshniak) [inline src/containers/layout/language-menu/LanguageMenu.constants.ts:10]: In such cases, it is better to name keys as `{value: '...', label: '...'}`.

**Review by byavdoshniak — APPROVED** [review]

File: src/containers/layout/language-menu/LanguageMenu.constants.ts (hunk 4 above)
Comment (D1nviz) [reply src/containers/layout/language-menu/LanguageMenu.constants.ts]: Changed

**Review by asya982 — APPROVED** [review]

**Pull request merged**